### PR TITLE
Updated DB Schema with dummy data and added User table APIs

### LINKED ISSUE #16: Create tables with dummy data
As a developer, I want to have a database populated with dummy data for the user table, Post table, and company table.

The user table should have the following fields:

- User ID (Primary Key)  - Integer
- First Name - String 
- Last Name - String
- Email - String
- Organization - String
- Title - String
- Password - String
- Current Location - String
- Current Company name - String
- Current Company ID - Integer
- Years of Experience Id- Integer

The Post table should have the following fields:

- Post Id (Primary Key) - Integer
- User Id (Foreign Key) - Integer
- Job link - String
- Target Company Id - Integer
- Target position - String
- About user - String
- Title - String
- Resume link - String

The Company table should have the following fields:

- Company Id (Primary Key) - Integer
- Company Name - String
- company email extension  - String

The Years_of_experience table should have the following fields:

- Years of Experience Id (Primary key) - Integer
- Description - String

Acceptance criteria:

- [x] Create User table with required fields as mentioned in the user story
- [x] Create Post table with required fields as mentioned in the user story
- [x] Create Company table with the required field as mentioned in the user story

### LINKED ISSUE #26: Expose API for User table
As a Frontend Developer, I want to receive all the users created properly formatted in JSON format.

As a Frontend Developer, I want to receive the user by email or id properly formatted in JSON format.

As a Frontend Developer, I want to update the User data by id with the required fields using JSON format.

As a Frontend Developer, I want to add a User using an HTTP request.

Acceptance Criteria:

- [x] Expose API to GET all users from the User table.
- [x] Expose APIs to GET a user by email or id
- [x] Expose API to Update the User from the table. 
- [x] Expose API to add a User to the table

### PR DESCRIPTION
**Pull request by shashank136**
Acceptance criteria:

- [x] Create User table with required fields as mentioned in the user story
- [x] Create Post table with required fields as mentioned in the user story
- [x] Create Company table with the required field as mentioned in the user story
- [x]  Expose API to GET all users from the User table.
- [x] Expose APIs to GET a user by email or id
- [x] Expose API to Update the User from the table. 
- [x] Expose API to add a User to the table

### CODE REVIEW

File: server/app/app.go [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 	"fmt"
 	"log"
 	"net/http"
-	"github.com/haxxorsid/referralboard-private/server/handler"
+	"strconv"
+
+	"github.com/gorilla/mux"
 	"github.com/haxxorsid/referralboard-private/server/config"
+	"github.com/haxxorsid/referralboard-private/server/models"
+	"github.com/haxxorsid/referralboard-private/server/services"
```
Comment (haxxorsid) [inline server/app/app.go]: sanitize these import statements

File: server/config/config.go [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
 		DB: &DBConfig{
-			Host:      "localhost",
-			User:      "siddhesh",
-			Password:  "toor",
-			Dbname:    "appdb",
-			Port:      "5432",
-			Sslmode:   "disable",
-			TimeZone:  "America/New_York",
+			Host:     "localhost",
+			User:     "shashank",
```
Comment (haxxorsid) [inline server/config/config.go]: remove credentials

Comment (shashank136) [reply server/config/config.go]: Removed the credentials

**Review by haxxorsid — APPROVED** [review]
Looks good now, approving.

**Review by bahetianjali — APPROVED** [review]
looks good.

**Review by Vineetkhadloya — APPROVED** [review]
Good, Approving !!

**Pull request merged**